routes/departments.go

```diff
@@ -46,3 +47,38 @@ func GetDepartments(ctx *gin.Context) {
 
 	ctx.JSON(http.StatusOK, resp)
 }
+
+// DeleteDepartment godoc
+// @Summary    delete a department
+// @Tags       departments
+// @Produce    json
+// @Param      id                              path        uuid                    true    "department id"
+// @Success    200							   {object}    HTTPSuccessResponse
+// @Failure    400                             {object}    HTTPErrorResponse
+// @Failure    404                             {object}    HTTPErrorResponse
+// @Router     /departments/{id}               [delete]
+func DeleteDepartment(ctx *gin.Context) {
+	_, logCtx := logging.GetRequestLogger(ctx)
+	db := models.GetDB(logCtx)
+
+	departmentId, _ := uuid.Parse(ctx.Param("id"))
+
+	err := db.Model(&models.Department{
+		DepartmentBase: models.DepartmentBase{
+			ID: departmentId,
+		},
+	}).Association("Rooms").Clear()
+
+	db.Delete(&models.Department{
```

Comment:
currently we ignore possible errors, we should handle them like this: https://github.com/helpwave/rest-api/blob/main/routes/rooms.go#L232